File: 2022/python/1_2.py
from queue import PriorityQueue

f = open("../data/in/1.in", "r")

curr_sum = 0
hq = PriorityQueue()

for line in f.read().splitlines():
    if line == "":
        hq.put(-curr_sum)
        curr_sum = 0
    else:
        curr_sum += int(line)

print(sum([-hq.get() for _ in range(3)])) 

File: 2022/rust/src/bin/1_2.rs
extern crate aoc2022;

use crate::aoc2022::utils::scanner::Scanner;
use std::collections::BinaryHeap;

fn main() {
    let input = std::fs::read_to_string("../data/in/1.in").unwrap();

    let mut sum = 0;
    let mut sums = BinaryHeap::new();

    for line in input.lines() {
        if line.is_empty() {
            sums.push(sum);
            sum = 0;
        } else {
            sum += line.parse::<i32>().unwrap();
        }
    }

    println!(
        "Top 3 sum: {}",
        (0..3).map(|_| sums.pop().unwrap()).sum::<i32>()
    );
}
